### PR TITLE
WINC-678: Upgrade to Operator SDK 1.15.0

### PR DESCRIPTION
**Pull request by alinaryan**
This PR updates Operator SDK in WMCO from v1.6.1 to latest v1.15.0. 

Followed upgrade steps from https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.15.0/

### CODE REVIEW

**Review by jrvaldes — COMMENTED** [review]
/lgtm

**Comment by sebsoto** [comment]
> Followed upgrade steps from https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.14.0/

@alinaryan  Please make sure you follow all upgrade instructions, from the current version, up to 1.14, not just the latest ones.

**Comment by alinaryan** [comment]
I skipped https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.11.0/#gov3-ansiblev1-helmv1-increase-resource-limits-for-controller-manager because we don't have cpu and resources fields in the config/manager/manger.yaml, but should I increase those limits elsewhere?

File: config/rbac/leader_election_role.yaml [hunk 1]
```diff
@@ -18,6 +16,10 @@ rules:
   - update
   - patch
   - delete
+  - apiGroups:
```
Comment (mansikulkarni96) [inline config/rbac/leader_election_role.yaml:19]: The indentation here does not align with other apiGroups and resources

Comment (aravindhp) [reply config/rbac/leader_election_role.yaml:19]: Which leads me to ask how was this tested?

**Comment by selansen** [comment]
> I skipped https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.11.0/#gov3-ansiblev1-helmv1-increase-resource-limits-for-controller-manager because we don't have cpu and resources fields in the config/manager/manger.yaml, but should I increase those limits elsewhere?

I think we need WMCO to be running even if it exceeds cpu/mem utilization(without it we will not be able to have widnows node in kube cluster. Not sure if this is omitted intentionally by design or is this option introduced in later part of operator SDK version. If we are adding some numbers then we need to make sure we have enoughdata point and add buffer so that WMCO pod will not be evicted/removed.

**Comment by aravindhp** [comment]
> > I skipped https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.11.0/#gov3-ansiblev1-helmv1-increase-resource-limits-for-controller-manager because we don't have cpu and resources fields in the config/manager/manger.yaml, but should I increase those limits elsewhere?
> 
> I think we need WMCO to be running even if it exceeds cpu/mem utilization(without it we will not be able to have widnows node in kube cluster. Not sure if this is omitted intentionally by design or is this option introduced in later part of operator SDK version. If we are adding some numbers then we need to make sure we have enoughdata point and add buffer so that WMCO pod will not be evicted/removed.

We need to be careful about the definition of "exceeds" as WMCO is not the most important pod on the node. I suggest we do some profiling and come up with the limits in a follow up PR.

File: config/rbac/leader_election_role.yaml (hunk 1 above)
Comment (alinaryan) [reply config/rbac/leader_election_role.yaml]: Fixed

Comment (alinaryan) [reply config/rbac/leader_election_role.yaml]: I was going off the indentation in the upgrade guide. When I did a local build it did not catch this. Thanks Mansi.

**Comment by selansen** [comment]
> > > I skipped https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.11.0/#gov3-ansiblev1-helmv1-increase-resource-limits-for-controller-manager because we don't have cpu and resources fields in the config/manager/manger.yaml, but should I increase those limits elsewhere?
> > 
> > 
> > I think we need WMCO to be running even if it exceeds cpu/mem utilization(without it we will not be able to have widnows node in kube cluster. Not sure if this is omitted intentionally by design or is this option introduced in later part of operator SDK version. If we are adding some numbers then we need to make sure we have enoughdata point and add buffer so that WMCO pod will not be evicted/removed.
> 
> We need to be careful about the definition of "exceeds" as WMCO is not the most important pod on the node. I suggest we do some profiling and come up with the limits in a follow up PR.

Agree. we should do some profiling with scale and measure CPU/Mem usage before come up with final number.
@alinaryan , let me know if you want me to create Jira ticket or you can create one and add it on our backlog .

**Comment by alinaryan** [comment]
>>>>I skipped https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.11.0/#gov3-ansiblev1-helmv1-increase->>>>resource-limits-for-controller-manager because we don't have cpu and resources fields in the >>>>config/manager/manger.yaml, but should I increase those limits elsewhere?

>>>I think we need WMCO to be running even if it exceeds cpu/mem utilization(without it we will not be able to have widnows >>>node in kube cluster. Not sure if this is omitted intentionally by design or is this option introduced in later part of operator >>>SDK version. If we are adding some numbers then we need to make sure we have enoughdata point and add buffer so that >>>WMCO pod will not be evicted/removed.

>>We need to be careful about the definition of "exceeds" as WMCO is not the most important pod on the node. I suggest we >>do some profiling and come up with the limits in a follow up PR.

>Agree. we should do some profiling with scale and measure CPU/Mem usage before come up with final number.
>@alinaryan , let me know if you want me to create Jira ticket or you can create one and add it on our backlog .

Sounds good! I will create one

**Comment by alinaryan** [comment]
/test aws-e2e-upgrade
Timed out provisioning Windows machine

**Review by saifshaikh48 — COMMENTED** [review]
I don't think the config/ changes will be reflected in the bundle image as is. Try `make bundle`

File: bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ metadata:
     olm.skipRange: '>=4.0.0 <5.0.0'
     operatorframework.io/cluster-monitoring: "true"
     operatorframework.io/suggested-namespace: openshift-windows-machine-config-operator
-    operators.operatorframework.io/builder: operator-sdk-v2.0.0+git
+    operators.operatorframework.io/builder: operator-sdk-v1.15.0+git
```
Comment (sebsoto) [inline bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml:14]: In the commit message `Update bundle image` isn't the correct terminology here, you're updating the bundle itself. We don't maintain the bundle image, that is something that a user can build using the bundle.

File: config/default/manager_auth_proxy_patch.yaml [hunk 3]
```diff
@@ -18,6 +18,7 @@ spec:
         - "--v=10"
         ports:
         - containerPort: 8443
+          protocol: TCP
```
Comment (sebsoto) [inline config/default/manager_auth_proxy_patch.yaml:21]: This change is going to require `make bundle` as well. I suggest moving this commit to before the `make bundle` commit.

**Review by sebsoto — CHANGES_REQUESTED** [review]
Good work @alinaryan 
Just two small changes requested

File: bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (sebsoto) [reply bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml:14]: To be clear `Update bundle directory` would be more accurate

File: config/default/manager_auth_proxy_patch.yaml (hunk 3 above)
Comment (alinaryan) [reply config/default/manager_auth_proxy_patch.yaml:21]: I reordered the commits, but nothing changed in the csv after running make bundle again. Should I have expected a change or was this change solely for proper commit ordering?

Comment (sebsoto) [reply config/default/manager_auth_proxy_patch.yaml:21]: If no change happened, that makes sense. Looks like this isnt being used at the moment by the operator.
In general all changes to config/ should be followed up by a make bundle.

Comment (alinaryan) [reply config/default/manager_auth_proxy_patch.yaml:21]: Makes sense! Thanks

**Comment by alinaryan** [comment]
/retest
DNS failures on Azure and AWS

**Comment by alinaryan** [comment]
\hold needs rebase

**Comment by alinaryan** [comment]
\hold

**Pull request merged**